hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/pipeline/transform/StreamSourceTransform.java

```diff
@@ -85,6 +86,9 @@ public void addToDag(Planner p) {
             String v1name = name();
             Vertex v1 = p.dag.newVertex(v1name, metaSupplierFn.apply(eventTimePolicy))
                              .localParallelism(localParallelism());
+            if (localParallelism() == LOCAL_PARALLELISM_USE_DEFAULT) {
```

Comment:
it would be better to explicitly set it to a local variable and then assign it to pv2 rather than changing parallelism of the transform, as done in TimestampTransform.